packages/remix/src/ssr/rootAuthLoader.ts

```diff
@@ -30,20 +36,17 @@ export const rootAuthLoader: RootAuthLoader = async (
   const frontendApi = process.env.CLERK_FRONTEND_API || opts.frontendApi;
   assertFrontendApi(frontendApi);
 
-  const { authData, showInterstitial } = await getAuthData(args.request, opts);
+  const { authData, showInterstitial, errorReason } = await getAuthData(args.request, opts);
 
   if (showInterstitial) {
-    throw json(
-      wrapClerkState({ __clerk_ssr_interstitial: showInterstitial, __frontendApi: frontendApi }), 
-      { status: 401 }
-    );
+    throw throwInterstitialJsonResponse({ frontendApi, errorReason });
```

Comment:
❓ 
Is the `throw` needed here since `throwInterstitialJsonResponse` throws on its own ?

Reply:
You're right, it's not **needed** per se, but I used it to further emphasize that we need to **throw** in order for the catch boundary to work - otherwise the SSR flow will break. It also helps TS infer that the execution stops there if the if is true, without using `return`

I was worried that wrapping the `throwInterstitialJsonResponse` into another function that uses a try/catch block during a future refactor could potentially break the flow.

Afaik immediately throwing a thrown error works as expected - but let me know if you have any objections